### PR TITLE
Make TaintIssuesSynchronizer ctor free threaded

### PR DESCRIPTION
**Pull request by bigfluffycookie**
part of https://github.com/SonarSource/sonarlint-visualstudio/issues/4861

### CODE REVIEW

File: src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs [hunk 1]
```diff
@@ -190,7 +185,16 @@ private void ClearStore()
             taintStore.Set(Enumerable.Empty<IAnalysisIssueVisualization>(), null);
         }
 
-        private void UpdateTaintIssuesUIContext(bool hasTaintIssues) =>
-            vsMonitorSelection.SetCmdUIContext(contextCookie, hasTaintIssues ? 1 : 0);
+        private void UpdateTaintIssuesUIContext(bool hasTaintIssues)
```
Comment (bigfluffycookie) [inline src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs:188]: Was able to move everything in here, I don't think calling getcmdcookie multiple times should be a problem

Comment (duncanp-sonar) [reply src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs:188]: Probably not... does VS return the same value for the cookie on each call to `GetCmdUIContextCookie`? If so, then it should be ok.

File: src/IssueViz.Security.UnitTests/Taint/TaintIssuesSynchronizerTests.cs [hunk 2]
```diff
@@ -54,11 +57,38 @@ public void MefCtor_CheckIsExported()
...
                 MefTestHelpers.CreateExport<IStatefulServerBranchProvider>(),
                 // The constructor calls the service provider so we need to pass a correctly-configured one
-                MefTestHelpers.CreateExport<SVsServiceProvider>(CreateServiceProvider()),
+                MefTestHelpers.CreateExport<IVsUIServiceOperation>(),
                 MefTestHelpers.CreateExport<IToolWindowService>(),
                 MefTestHelpers.CreateExport<ILogger>());
         }
 
+        [TestMethod]
+        public void MefCtor_DoesNotCallAnyServices_BesidesExpected()
```
Comment (duncanp-sonar) [inline src/IssueViz.Security.UnitTests/Taint/TaintIssuesSynchronizerTests.cs:66]: `_BesidesExpected` - the constructor doesn't call any services, does it?

**Review by duncanp-sonar — COMMENTED** [review]
Looks good. Just a couple of questions...

File: src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs (hunk 1 above)
Comment (bigfluffycookie) [reply src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs:188]: Here it says "Note Once a GUID is registered through this method, the GUID remains registered until the IDE exits." So from my understanding yes it always returns the same one

https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.interop.ivsmonitorselection.getcmduicontextcookie?view=visualstudiosdk-2022

**Pull request merged**